### DISCUSSION TITLE
Underlined words with no definition?

### DISCUSSION THREAD

**Discussion by mshoumans** [xxyzz/WordDumb#131] [General]
First of all, thank you so much for the tool. I am using koreader on android so below are my questions

Some words in the book are underlined but dont have a definition above them. I saw in lemma that they have definition, so what could be issue? and they are sometimes simple words.

Second issue is in epub in koreader, is there a way to turn off unlined words and their definition like kindle? or its permanent per book?

**Comment by xxyzz** [comment]
Please provide more details about the no definition of underlined words bug, what's your configuration and upload the book if you only find this bug on certain books.

You can't turn off the word wise and x-ray in EPUB books, they are added to the book's xhtml file.

**Comment by mshoumans** [comment]
![WordDump2](https://github.com/xxyzz/WordDumb/assets/21187891/a7f0a360-75ad-4090-a781-ad15c7409a0c)
![WordDump1](https://github.com/xxyzz/WordDumb/assets/21187891/1dc7e108-dfea-40e6-b31e-feeabfc788c1)

Settings above and look how the word shows up below, most are missing definition above word e.g. guffaw vs word sword  or initiated. This happens on multiple books btw
![20230603_171904](https://github.com/xxyzz/WordDumb/assets/21187891/2d1cd082-0c2f-4d86-b679-30b6c3da26fd)

**Comment by mshoumans** [comment]
Let me know how you prefer i upload book or images in future. Also how do i see x-ray stuff in epub, can you explain how to find it?

**Reply by xxyzz** [reply]
If a word's definition is too long it won't be shown above the word, you can click the word to see its gloss. X-Ray entries are also underlined, you have to click the underlined word to see the person or location summary. Both Word Wise and X-Ray are added as footnotes to EPUB books, you may need to change koreader's footnotes settings.

**Reply by moe123s** [reply]
Thank you. But I didn't think for example guffaw definition was too long What is the word size limit to display ? And is it possible in future to maybe add option to keep word definition even if considered long?

Also problem with footnote settings is they overwhelm a page like a page could have one paragraph with rest of the page footnotes, that's why I turned footnotes off. Is it expected that page could be overwhelmed by footnotes? Also wish the footnotes were brief, like when I click a word want to see max 1 or 2 definitions not an entire article on the word

**Reply by xxyzz** [reply]
Gloss won't be added if it's length ratio to word is greater than 3: https://github.com/xxyzz/WordDumb/blob/0f195cbfab11721042a72a43dfc86845f684f677/epub.py#L297-L302

The book won't be readable if long glosses are shown. Because then you will have most lines have only one or two words.

You should use the pop up footnote.
